cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -1034,50 +1036,53 @@ impl FromStr for CatalogPackage {
             None => (descriptor.to_string(), None),
         };
 
-        let install_id = {
-            let mut install_id = None;
-            let mut cur = String::new();
+        let install_id = install_id_from_attr_path(&attr_path, descriptor)?;
 
-            let mut start_quote = None;
+        Ok(Self {
+            id: install_id,
+            pkg_path: attr_path,
+            version,
+        })
+    }
+}
 
-            for (n, c) in attr_path.chars().enumerate() {
-                match c {
-                    '.' if start_quote.is_none() => {
-                        let _ = install_id.insert(std::mem::take(&mut cur));
-                    },
-                    '"' if start_quote.is_some() => start_quote = None,
-                    '"' if start_quote.is_none() => start_quote = Some(n),
-                    other => cur.push(other),
-                }
-            }
+/// parses a dot-separated attribute path into
+fn install_id_from_attr_path(attr_path: &str, descriptor: &str) -> Result<String, ManifestError> {
+    let mut install_id = None;
+    let mut cur = String::new();
 
-            if start_quote.is_some() {
-                return Err(ManifestError::MalformedStringDescriptor {
-                    msg: "unclosed quote".to_string(),
-                    desc: descriptor.to_string(),
-                });
-            }
+    let mut start_quote = None;
 
-            if !cur.is_empty() {
-                let _ = install_id.insert(cur);
-            }
+    for (n, c) in attr_path.chars().enumerate() {
+        match c {
+            '.' if start_quote.is_none() => {
+                let _ = install_id.insert(std::mem::take(&mut cur));
+            },
+            '"' if start_quote.is_some() => start_quote = None,
+            '"' if start_quote.is_none() => start_quote = Some(n),
+            other => cur.push(other),
+        }
+    }
 
-            if install_id.is_none() {
-                return Err(ManifestError::MalformedStringDescriptor {
-                    msg: "attribute path is empty".to_string(),
-                    desc: descriptor.to_string(),
-                });
-            }
+    if start_quote.is_some() {
+        return Err(ManifestError::MalformedStringDescriptor {
+            msg: "unclosed quote".to_string(),
+            desc: descriptor.to_string(),
+        });
+    }
 
-            install_id.unwrap()
-        };
+    if !cur.is_empty() {
+        let _ = install_id.insert(cur);
+    }
 
-        Ok(Self {
-            id: install_id,
-            pkg_path: attr_path,
-            version,
-        })
+    if install_id.is_none() {
+        return Err(ManifestError::MalformedStringDescriptor {
+            msg: "attribute path is empty".to_string(),
+            desc: descriptor.to_string(),
+        });
     }
+
+    Ok(install_id.unwrap())
```

Comment:
nit: I would prefer a match or if let to `is_none()`/`unwrap()`

Reply:
Fixed